### PR TITLE
Structlog dep fix

### PR DESCRIPTION
**Pull request by luciipak**
`We` were using threadlocal module in structlog (https://www.structlog.org/en/23.2.0/thread-local.html) to manage binding of relevant request variables (eg customer id, user id, upload/job ids, ....) to the logger, which is now deprecated in favor of context module (https://www.structlog.org/en/stable/contextvars.html).

With this change as is, context variables bound in the request handler are not available to middleware handler which needs them. (see example https://github.com/tiangolo/fastapi/issues/4696).
There is also now a relevant warning in the structlog docs itself for fastapi:

```
Since the storage mechanics of your context variables is different for each concurrency method, they are isolated from each other.

This can be a problem in hybrid applications like those based on [starlette](https://www.starlette.io/) (this [includes FastAPI](https://github.com/tiangolo/fastapi/discussions/5999)) where context variables set in a synchronous context don’t appear in logs from an async context and vice versa.
```

The fix in this pr uses `startlette-context` to manage context through entire request-response cycle to store relevant request variables for use in the middleware logs and exceptions.

### CODE REVIEW

File: frontend/pulse3d/src/components/basicWidgets/ButtonWidget.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  position: relative;
+  width: ${props.width || "100%"};
+  height: ${props.height || "60px"};
+  top: ${props.top || "0px"};
+  left: ${props.left || "0px"};
+`
+);
+
+// TODO should this automatically disable itself if props.inProgress === true ?
+const ButtonWidget = (props) => {
```
Comment (tannermpeterson) [inline frontend/pulse3d/src/components/basicWidgets/ButtonWidget.js:46]: What is this widget for?

Comment (luciipak) [reply frontend/pulse3d/src/components/basicWidgets/ButtonWidget.js:46]: somewhere this file got removed and needs to be capitalized, do you have it as buttonWidget?

Comment (tannermpeterson) [reply frontend/pulse3d/src/components/basicWidgets/ButtonWidget.js:46]: I'm not sure

Comment (luciipak) [reply frontend/pulse3d/src/components/basicWidgets/ButtonWidget.js:46]: Well regardless it'll need to be added back to RC-12-07-23

**Review by tannermpeterson — APPROVED** [review]

**Pull request merged**